cmd/main.go

```diff
@@ -5,163 +5,196 @@ package main
-}
-
-func parseCode(code string) (codes.Code, error) {
-	ret, err := codes.ToCode(code)
-	if err != nil {
-		return 0, err
-	}
-	switch ret {
-	case codes.GET, codes.POST, codes.PUT, codes.DELETE:
-		return ret, nil
-	default:
-		return 0, errCodeNotSupported
-	}
-}
-
-func printMsg(m *pool.Message) {
-	if m != nil {
-		log.Printf("\nMESSAGE:\n %v", m)
+func main() {
+	rootCmd := &cobra.Command{
+		Use:   "coap-cli <method> <URL> [options]",
+		Short: "CLI for CoAP",
+	}
+
+	getCmd := &cobra.Command{
+		Use:   "get <url>",
+		Short: "Perform a GET request on a COAP resource",
+		Example: "coap-cli get channels/0bb5ba61-a66e-4972-bab6-26f19962678f/messages/subtopic -a 1e1017e6-dee7-45b4-8a13-00e6afeb66eb -H localhost -p 5683 -O 17,50 -o \n" +
+			"coap-cli get channels/0bb5ba61-a66e-4972-bab6-26f19962678f/messages/subtopic --auth 1e1017e6-dee7-45b4-8a13-00e6afeb66eb --host localhost --port 5683 --options 17,50 --observe",
+		Run: runCmd(codes.GET),
+	}
+	getCmd.Flags().BoolVarP(&observe, "observe", "o", false, "Observe resource")
+
+	putCmd := &cobra.Command{
+		Use:   "put <url>",
+		Short: "Perform a PUT request on a COAP resource",
+		Example: "coap-cli put /test -H coap.me -p 5683 -c 50 -d 'hello, world'\n" +
+			"coap-cli put /test --host coap.me --port 5683 --content-format 50 --data 'hello, world'",
+		Run: runCmd(codes.PUT),
+	}
+	putCmd.Flags().StringVarP(&data, "data", "d", "", "Data")
+
+	postCmd := &cobra.Command{
+		Use:   "post <url>",
+		Short: "Perform a POST request on a COAP resource",
+		Example: "coap-cli post channels/0bb5ba61-a66e-4972-bab6-26f19962678f/messages/subtopic -a 1e1017e6-dee7-45b4-8a13-00e6afeb66eb  -H localhost -p 5683 -c 50 -d 'hello, world'\n" +
+			"coap-cli post channels/0bb5ba61-a66e-4972-bab6-26f19962678f/messages/subtopic  --auth 1e1017e6-dee7-45b4-8a13-00e6afeb66eb --host localhost --port 5683 --content-format 50 --data 'hello, world'",
+		Run: runCmd(codes.POST),
+	}
+	postCmd.Flags().StringVarP(&data, "data", "d", "", "Data")
+
+	deleteCmd := &cobra.Command{
+		Use:   "delete <url>",
+		Short: "Perform a DELETE request on a COAP resource",
+		Example: "coap-cli delete /test -H coap.me -p 5683 -c 50 -d 'hello, world' -O 17,50\n" +
+			"coap-cli delete /test --host coap.me --port 5683 --content-format 50 --data 'hello, world' --options 17,50",
+		Run: runCmd(codes.DELETE),
+	}
+	deleteCmd.Flags().StringVarP(&data, "data", "d", "", "Data")
+
+	rootCmd.AddCommand(getCmd, putCmd, postCmd, deleteCmd)
+	rootCmd.PersistentFlags().StringVarP(&host, "host", "H", "localhost", "Host")
+	rootCmd.PersistentFlags().StringVarP(&port, "port", "p", "5683", "Port")
+	rootCmd.PersistentFlags().StringVarP(&auth, "auth", "a", "", "Auth")
+	rootCmd.PersistentFlags().IntVarP(&contentFormat, "content-format", "c", 50, "Content format")
+	rootCmd.PersistentFlags().StringArrayVarP(&options, "options", "O", []string{}, "Options")
+	rootCmd.PersistentFlags().Uint64VarP(&keepAlive, "keep-alive", "K", 60, "Keep alive interval")
```

Comment:
For KeepAlive. something like this will be more informative to user
```
Send a ping after interval seconds of inactivity. If not specified (or 0), keep-alive is disabled (default).
```